File: mountain/src/main/java/com/santeut/mountain/dto/response/AllCourseResponse.java
package com.santeut.mountain.dto.response;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AllCourseResponse {

  public List<CourseSummary> course;

  @Data
  @NoArgsConstructor
  public static class CourseSummary {

    public int courseId;
    public double distance;
    public List<LocationData> locationDataList;

    public CourseSummary(int courseId, double distance, List<LocationData> locationDataList) {
      this.courseId = courseId;
      this.distance = Math.round(distance * 100) / 100.0;
      this.locationDataList = locationDataList;
    }
  }

}

File: android/app/src/main/java/com/santeut/data/model/response/AllcourseResponse.kt
package com.santeut.data.model.response

import com.google.gson.annotations.SerializedName

data class AllCourseResponse(
    @SerializedName("course") val course: List<CourseDetailResponse>
)

data class CourseDetailResponse(
    @SerializedName("courseId") val courseId: Int,
    @SerializedName("distance") val distance: Double,
    @SerializedName("locationDataList") val locationDataList: List<LocationDataResponse>
) {
    val validLocationDataList: List<LocationDataResponse>
        get() = locationDataList.filter { !it.lat.isNaN() && !it.lng.isNaN() }
}

data class LocationDataResponse(
    @SerializedName("lat") val lat: Double,
    @SerializedName("lng") val lng: Double
)